src/custom/state/claim/hooks/index.ts

```diff
@@ -94,3 +114,160 @@ export function useUserUnclaimedAmount(account: string | null | undefined): Curr
 
   return CurrencyAmount.fromRawAmount(vCow, JSBI.BigInt(totalAmount))
 }
+
+/**
+ * Hook that returns the claimCallback
+ *
+ * Different from the original version, the returned callback takes as input a list of ClaimInputs,
+ * which is an object of the claim index and the amount being claimed.
+ *
+ * @param account
+ */
+export function useClaimCallback(account: string | null | undefined): {
+  claimCallback: (claimInputs: ClaimInput[]) => Promise<string | undefined>
+} {
+  // get claim data for given account
+  const { chainId, account: connectedAccount } = useActiveWeb3React()
+  const claims = useUserAvailableClaims(account)
+  const vCowContract = useVCowContract()
+
+  // used for popup summary
+  const addTransaction = useTransactionAdder()
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  const claimCallback = useCallback(
+    async function (claimInput: ClaimInput[]) {
+      if (
+        claims.length === 0 ||
+        claimInput.length === 0 ||
+        !account ||
+        !connectedAccount ||
+        !chainId ||
+        !vCowContract ||
+        !vCowToken
+      ) {
+        throw new Error("Not initialized, can't claim")
+      }
+
+      const { args, totalClaimedAmount, value } = _getClaimManyArgs({ claimInput, claims, account, connectedAccount })
+
+      if (!args) {
+        throw new Error('There were no valid claims selected')
+      }
+
+      const vCowAmount = CurrencyAmount.fromRawAmount(vCowToken, totalClaimedAmount)
+
+      return vCowContract.estimateGas['claimMany'](...args, { value }).then((estimatedGas) =>
+        vCowContract
+          .claimMany(...args, {
+            from: connectedAccount,
+            gasLimit: calculateGasMargin(chainId, estimatedGas),
+            value,
+          })
+          .then((response: TransactionResponse) => {
+            addTransaction({
+              hash: response.hash,
+              summary: `Claimed ${formatSmart(vCowAmount)} vCOW`,
+              claim: { recipient: account },
+            })
+            return response.hash
+          })
+      )
+    },
+    [account, addTransaction, chainId, claims, connectedAccount, vCowContract, vCowToken]
+  )
+
+  return { claimCallback }
+}
+
+type GetClaimManyArgsParams = {
+  claimInput: ClaimInput[]
+  claims: UserClaims
+  account: string
+  connectedAccount: string
+}
+
+function _getClaimManyArgs({ claimInput, claims, account, connectedAccount }: GetClaimManyArgsParams): {
+  args: [number[], ClaimType[], string[], string[], string[], string[][], string[]] | undefined
+  totalClaimedAmount: JSBI
+  value: string | undefined
+} {
+  const indices: number[] = []
+  const claimTypes: ClaimType[] = []
+  const claimants: string[] = []
+  const claimableAmounts: string[] = []
+  const claimedAmounts: string[] = []
+  const merkleProofs: string[][] = []
+  const sendEth: string[] = []
```

Comment:
can you comment on what theese are ? e.g what is sendEth?

Reply:
Added a comment pointing to where they came from: https://github.com/gnosis/gp-v2-token/blob/main/src/contracts/mixins/MerkleDistributor.sol#L123